docs/reference/migration/migrate_7_10.asciidoc

```diff
@@ -43,9 +62,85 @@ The `name` property is now required to create or grant an API key.
 To avoid errors, specify the `name` property when creating or granting API keys.
 ====
 
+[discrete]
+[[breaking_710_java_changes]]
+==== Java changes
+
+[[supplier-searchlookup-arg]]
+.The `MappedFieldType#fielddataBuilder` method now accepts a `Supplier<SearchLookup>` argument.
+[%collapsible]
+====
+*Details* +
+To support future feature development, the existing
+`MappedFieldType#fielddataBuilder` method now accepts a new
+`Supplier<SearchLookup>` argument.
+
+*Impact* +
+If you develop or maintain a mapper plugin, update your implementation of the
+`MappedFieldType#fielddataBuilder` method to accommodate the new signature.
+====
+
+[discrete]
+[[breaking_710_networking_changes]]
+==== Networking changes
+
+[keep-idle-and-keep-internal-limits]
+.The `*.tcp.keep_idle` and `*.tcp.keep_interval` settings are now limited to `300` seconds.
+[%collapsible]
+====
+*Details* +
+The `{network,transport,http}.tcp.keep_idle` and
+`{network,transport,http}.tcp.keep_interval` settings now have a maximum
+value of `300` seconds, equivalent to 5 minutes.
+
+*Impact* +
+If specified, ensure the `{network,transport,http}.tcp.keep_idle` and
+`{network,transport,http}.tcp.keep_interval` settings do not exceed `300`
+seconds. Setting `{network,transport,http}.tcp.keep_idle` or
+`{network,transport,http}.tcp.keep_interval` to a value greater than `300`
+seconds in `elasticsearch.yml` will result in an error on startup.
+====
+
+[discrete]
+[[breaking_710_search_changes]]
+==== Search changes
+
+[[max-doc-value-field-search-limits]]
+.The `index.max_docvalue_fields_search` setting now limits doc value fields returned by `inner_hits` or the `top_hits` aggregation.
+[%collapsible]
+====
+*Details* +
+The `index.max_docvalue_fields_search` setting limits the number of doc value
+fields retrieved by a search. Previously, this setting applied only to doc value
+fields returned by the `docvalue_fields` parameter in a top-level search. The
+setting now also applies to doc value fields returned by an `inner_hits` section
+or `top_hits` aggregation.
+
+*Impact* +
+If you use `inner_hits` or the `top_hits` aggregation, ensure
+`index.max_docvalue_fields_search` is configured correctly for your use case.
+====
+
+//end::notable-breaking-changes[]
```

Comment:
Should we rework the Breaking changes section of Install and Upgrade guide as a migration guide as well?
https://www.elastic.co/guide/en/elastic-stack/7.10/elastic-stack-breaking-changes.html

I imagine those users would want to know about deprecations as well.

Reply:
Yeah, I think framing the breaking changes in terms of migration makes sense. I'll open up a follow on PR for that once I've merged the ES changes.